pages/index.tsx

```diff
@@ -8,18 +8,28 @@ import LoginWithSMS from '../components/LoginWithSMS';
 import { LoginMethod } from '../lib/types';
 import LoginEntryPoint from '../components/LoginEntryPoint';
 
-const REDIRECT_URL_BASE = process.env.USE_VERCEL == 'true' ?
-  `https://${process.env.VERCEL_URL}/api/authenticate_magic_link` :
-  'http://localhost:3000/api/authenticate_magic_link';
+// Set the URL base for redirect URLs. The three cases are as follows:
+// 1. Running locally via `vercel dev`; VERCEL_URL will contain localhost, but will not be https.
+// 2. Deploying via Vercel; VERCEL_URL will be generated on runtime and use https.
+// 3. Running locally via `npm run dev`; VERCEL_URL will be undefined and the app will be at localhost.
+let REDIRECT_URL_BASE = '';
 
+if (process.env.VERCEL_URL != undefined &&
+    process.env.VERCEL_URL.includes('localhost')) {
+  REDIRECT_URL_BASE = 'http://localhost:3000';
+} else if (process.env.VERCEL_URL != undefined) {
+  REDIRECT_URL_BASE = `https://${process.env.VERCEL_URL}`;
```

Comment:
I'm surprised that an env var called `VERCEL_URL` isn't actually the whole URL. Can we add that to the block comment?

Reply:
Added a note in 790cfcb